spec/index.bs

```diff
@@ -7633,26 +7652,51 @@ dictionary GPUComputePipelineDescriptor
+                    except capturing any errors as |error|, rather than dispatching them to the device.
+
+                1. When either of the following events occur (whichever comes first),
+                    or if either has already occurred:
+
+                    - The [=device timeline=] becomes informed that [=pipeline creation=] has
+                        completed for |pipeline| (successfully or unsuccessfully).
+                    - |this| [=becomes lost=].
 
-                1. When |pipeline| is ready to be used or has been [$invalidated$]:
-                    1. Let |valid| be |pipeline|'s [$valid$] state.
-                    1. Issue the subsequent steps on <var data-timeline=content>contentTimeline</var>.
+                    Then issue the subsequent steps on the [=device timeline=] of |this|.
             </div>
-            <div data-timeline=content>
-                [=Content timeline=] steps:
+            <div data-timeline=device>
+                [=Device timeline=] steps:
 
-                1. If |valid| is...
-                    <dl class=switch>
-                        : `true`
-                        :: [=Resolve=] |promise| with |pipeline|.
-                        : `false` due to an [$internal error$]
-                        :: [=Reject=] |promise| with a {{GPUPipelineError}} with
+                1. If |this| is [$invalid|lost$] or |pipeline| is [$valid$],
+                    issue the following steps on |contentTimeline|, and return.
+
+                    <div data-timeline=content>
+                        [=Content timeline=] steps:
+
+                        1. [=Resolve=] |promise| with |pipeline|.
+                    </div>
+
+                    Note: No errors are generated after device loss. See [[#errors-and-debugging]].
+
+                1. If |pipeline| is [$invalid$] and |error| is an [$internal error$],
+                    issue the following steps on |contentTimeline|, and return.
+
+                    <div data-timeline=content>
+                        [=Content timeline=] steps:
+
+                        1. [=Reject=] |promise| with a {{GPUPipelineError}} with
                             {{GPUPipelineErrorInit/reason}} {{GPUPipelineErrorReason/"internal"}}.
-                        : `false` due to a [$validation error$]
-                        :: [=Reject=] |promise| with a {{GPUPipelineError}} with
+                    </div>
+
+                1. If |pipeline| is [$invalid$] and |error| is a [$validation error$],
+                    issue the following steps on |contentTimeline|, and return.
+
+                    <div data-timeline=content>
+                        [=Content timeline=] steps:
+
+                        1. [=Reject=] |promise| with a {{GPUPipelineError}} with
                             {{GPUPipelineErrorInit/reason}} {{GPUPipelineErrorReason/"validation"}}.
-                    </dl>
+                    </div>
             </div>
```

Comment:
All of this text is exactly the same between render and compute. Could try to move it into an algorithm - wasn't sure.